source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -700,16 +702,97 @@ def apply_final_journey_filters_post_finalize(response_list, request):
+        self.minimum = minimum
+        self.maximum = maximum
 
-def keep_olympics_journeys(responses, request):
-    keep_olympics_journeys = request.get("_keep_olympics_journeys", False)
-    if keep_olympics_journeys:
-        for response in responses:
-            for journey in response.journeys:
-                if 'to_delete' in journey.tags and 'olympics' in journey.tags:
-                    journey.tags.remove("to_delete")
+    def includes(self, n):
+        return self.minimum <= n <= self.maximum
+
+
+def filter_olympics_journeys(responses, request):
+    if request.get('debug', False):
+        return
+
+    if not request.get("_keep_olympics_journeys", False):
+        return
+
+    intervals = {
+        # 9
+        "stop_point:IDFM:monomodalStopPlace:58498": Interval(0, 4240),
+        # 6
+        "stop_point:IDFM:22041": Interval(4240, 5606),
+        "stop_point:IDFM:463281": Interval(4240, 5606),
+        # 4
+        "stop_point:IDFM:412988": Interval(5606, float('inf')),
+        "stop_point:IDFM:412987": Interval(5606, float('inf')),
+    }
+
+    attractivity = {
+        "stop_point:IDFM:monomodalStopPlace:58498": 9,
+        "stop_point:IDFM:22041": 6,
+        "stop_point:IDFM:463281": 6,
+        "stop_point:IDFM:412988": 4,
+        "stop_point:IDFM:412987": 4,
+    }
+
+    olympics_durations = [j.duration for r in responses for j in r.journeys if 'olympics' in j.tags]
+    average_duration = sum(olympics_durations) / float(len(olympics_durations))
+
+    retained_journey = None
+
+    found_sps = []
+    for r in responses:
+        for j in r.journeys:
+            if 'olympics' not in j.tags:
+                continue
+            pt_extremity = get_journey_pt_extremity(j, request.get('clockwise'))
+            if pt_extremity.uri not in found_sps:
+                found_sps.append(pt_extremity.uri)
+            interval = intervals.get(pt_extremity.uri)
+            print(pt_extremity.uri, interval, average_duration)
+            if not interval.includes(average_duration):
+                j.tags.append("to_delete")
+            else:
+                if retained_journey is None:
+                    retained_journey = j
+                elif retained_journey.duration > j.duration:
+                    retained_journey.tags.append("to_delete")
+                    retained_journey = j
+                else:
+                    j.tags.append("to_delete")
+
+    if retained_journey is not None:
+        return
+
+    found_sps.sort(key=lambda sp: attractivity[sp])
+    best_attractivity = found_sps[-1]
+    for r in responses:
+        for j in r.journeys:
+            if 'olympics' not in j.tags:
+                continue
+            pt_extremity = get_journey_pt_extremity(j, request.get('clockwise'))
+            if pt_extremity.uri != best_attractivity:
+                continue
+            if retained_journey is None or retained_journey.duration > j.duration:
+                retained_journey = j
```

Comment:
I'm not sure that I understand this second pass. First pass don't remove any journey so we still have at least one journey for each olymic game's site. And because they're all present, `found_sps` will contain all the stop areas associated with this olympic game site. So this second pass will only select the journey which go through the best attractive olympic game's site... which is not what we want? (I'm trying to write what I understood so you can identify better where my reasoning is flawed :smile: )

Reply:
yes! totally